fe/fe-core/src/main/java/org/apache/doris/planner/external/hudi/HudiScanNode.java

```diff
@@ -0,0 +1,231 @@
+        fileDesc.setDeltaFilePaths(hudiSplit.getHudiDeltaLogs());
+        tableFormatFileDesc.setHudiParams(fileDesc);
+        rangeDesc.setTableFormatParams(tableFormatFileDesc);
+    }
+
+    @Override
+    public List<Split> getSplits() throws UserException {
+        // TODO expressions
+        // List<Expression> expressions = new ArrayList<>();
+
+        // for (Expr conjunct : conjuncts) {
+        //     Expression expression = null;
+        //     if (expression != null) {
+        //         expressions.add(expression);
+        //     }
+        // }
+
+        HMSExternalTable targetTable = (HMSExternalTable) source.getTargetTable();
+        HoodieTableMetaClient hudiClient = HiveMetaStoreClientHelper.getHudiClient(targetTable);
+
+        hudiClient.reloadActiveTimeline();
+
+        String partitionName = getHudiPartitionName(targetTable, hudiClient);
+
+        List<Split> splits = new ArrayList<>();
+
+        String basePath = targetTable.getRemoteTable().getSd().getLocation();
+        String inputFormat = targetTable.getRemoteTable().getSd().getInputFormat();
+        String serdeLib = targetTable.getRemoteTable().getSd().getSerdeInfo().getSerializationLib();
+
+        TableSchemaResolver schemaUtil = new TableSchemaResolver(hudiClient);
+        Schema hudiSchema;
+        try {
+            hudiSchema = HoodieAvroUtils.createHoodieWriteSchema(schemaUtil.getTableAvroSchema());
+        } catch (Exception e) {
+            throw new RuntimeException("Cannot get hudi table schema.");
+        }
+
+        List<String> columnNames = new ArrayList<>();
+        List<String> columnTypes = new ArrayList<>();
+        List<FieldSchema> allFields = targetTable.getRemoteTable().getSd().getCols();
+        allFields.addAll(targetTable.getRemoteTable().getPartitionKeys());
+
+        for (Schema.Field hudiField : hudiSchema.getFields()) {
+
+            String columnName = hudiField.name().toLowerCase(Locale.ROOT);
+            Optional<FieldSchema> field = allFields.stream()
+                    .filter(f -> f.getName().equals(columnName)).findFirst();
+            if (!field.isPresent()) {
+                String errorMsg = String.format("Hudi column %s not exists in hive metastore.", hudiField.name());
+                throw new IllegalArgumentException(errorMsg);
+            }
+            columnNames.add(columnName);
+            String columnType = HudiExternalTable.fromAvroHudiTypeToHiveTypeString(hudiField.schema());
+            columnTypes.add(columnType);
+        }
+
+        try {
+            String globPath = String.format("%s/%s/*", hudiClient.getBasePath(), partitionName);
+            List<FileStatus> statuses = FSUtils.getGlobStatusExcludingMetaFolder(hudiClient.getRawFs(),
+                    new Path(globPath));
+
+            HoodieTimeline timeline = hudiClient.getCommitsAndCompactionTimeline().filterCompletedInstants();
+            Option<HoodieInstant> latestInstant = timeline.lastInstant();
```

Comment:
So currently only support MOR from the last snapshots. Can we reserve interfaces to support `Time Travel Query` and `Incremental query`.